File: Codeforces/Gym/gym100499/i.py
import math
power10 = [0 for _ in range(16)]
power10[0] = 1
for i in range(1, 16):
    power10[i] = power10[i - 1] * 10
numTest = int(input().strip())
for testCase in range(1, numTest + 1):
    print("Case #", end = '')
    print(testCase, end = '')
    print(": ", end = '')
    frac = input()
    a = 0
    posDot = -1
    for i in range(len(frac)):
        if frac[i] == '.':
            posDot = i
            break
        else:
            a = a * 10 + ord(frac[i]) - 48
    if posDot == -1:
        print(a, end = '')
        print('/', end = '')
        print(1)
    else:
        b = 0 
        posBrack = -1
        for i in range(posDot + 1, len(frac)):
            if frac[i] == '(':
                posBrack = i
                break 
            else:
                b = b * 10 + ord(frac[i]) - 48
        if posBrack == -1:
            deno = b 
            nume = power10[len(frac) - posDot - 1]
            g = math.gcd(deno, nume)
            deno //= g 
            nume //= g
            deno = deno + a * nume 
            print(deno, end = '')
            print('/', end = '')
            print(nume)
        else:
            tmpDeno = b 
            tmpNume = power10[posBrack - posDot - 1]
            g = math.gcd(tmpDeno, tmpNume)
            tmpDeno //= g 
            tmpNume //= g 
            x = 0 
            for i in range(posBrack + 1, len(frac) - 1):
                x = x * 10 + ord(frac[i]) - 48
            y = power10[len(frac) - 2 - posBrack]
            y -= 1 
            y *= power10[posBrack - posDot - 1]
            g = math.gcd(x, y)
            x //= g 
            y //= g 
            x = x * tmpNume + y * tmpDeno
            y *= tmpNume
            g = math.gcd(x, y)
            x //= g 
            y //= g
            x = x + a * y
            print(x, end = '')
            print('/', end = '')
            print(y)
            

File: Codeforces/Gym/gym100499/i.cpp
#include <iostream>
#include <string> 
#include <algorithm>
#include <math.h> 
int numTest; 
std::string frac; 
long long power10[16]; 

int main () {
  power10[0] = 1LL;
  for (int i = 1; i <= 15; i++) {
    power10[i] = power10[i - 1] * 10LL; 
  }
  std::cin >> numTest;
  for (int testCase = 1; testCase <= numTest; testCase++) {
    std::cout << "Case #" << testCase << ": "; 
    std::cin >> frac; 
    long long a = 0;
    int posDot = -1; 
    for (int i = 0; i < (int) frac.size(); i++) {
      if (frac[i] == '.') {
        posDot = i; 
        break; 
      }
      else {
        a = a * 10LL + frac[i] - '0'; 
      }
    }
    long long deno = -1, nume = -1;
    if (posDot == -1) {
      std::cout << a << '/' << 1 << std::endl; 
    }   
    else {
      long long b = 0; 
      int posBrack = -1; 
      for (int i = posDot + 1; i < (int) frac.size(); i++) {
        if (frac[i] == '(') {
          posBrack = i; 
          break; 
        }
        else {
          b = b * 10LL + frac[i] - '0'; 
        }
      }
      if (posBrack == -1) {
        deno = b;
        nume = power10[(int) frac.size() - posDot - 1];
        long long gcd = std::__gcd(deno, nume);
        deno /= gcd;
        nume /= gcd;
        deno += a * nume;
        std::cout << deno << '/' << nume << std::endl;   
      }
      else {
        long long tmpDeno = b; 
        long long tmpNume = power10[posBrack - posDot - 1];
        long long gcd = std::__gcd(tmpDeno, tmpNume);
        tmpDeno /= gcd;
        tmpNume /= gcd;
        long long x = 0LL; 
        for (int i = posBrack + 1; i < (int) frac.size() - 1; i++) {
          x = x * 10LL + frac[i] - '0'; 
        } 
        long long y = power10[(int) frac.size() - 2 - posBrack];  
        y--; 
        y *= power10[posBrack - posDot - 1];
        gcd = std::__gcd(x, y);
        x /= gcd;
        y /= gcd;
        x = x * tmpNume + tmpDeno * y;
        y *= tmpNume;
        gcd = std::__gcd(x, y);
        x /= gcd;
        y /= gcd;
        x += a * y;
        std::cout << x << "/" << y << std::endl;   
      }
    }
  }

  return 0; 
}